oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticDocument.java

```diff
@@ -79,6 +81,7 @@ public class ElasticDocument {
         this.fulltext = new LinkedHashSet<>();
         this.suggest = new LinkedHashSet<>();
         this.spellcheck = new LinkedHashSet<>();
+        this.nullProperties = new LinkedHashSet<>();
         this.properties = new HashMap<>();
```

Comment:
Just a side comment, not related to this PR. All the other datastructures used for the fields in this class have a well-defined iteration order, except this one. Could this be an issue? If so, for consistency with the other fields, we could use a `LinkedHashMap` here. Or if the order is not important, then we can use `HashSet` for the others instead of `LinkedHashSet`.

Reply:
the properties using `LinkedHashSet` are multi-valued stored in a single ES field. Using `HashSet` will not preserve the order. This can slightly affect relevance but mostly will make it harder to debug (when looking at the document source).
`properties` is an `HashMap` because it gets _extracted_ into multiple fields, so the order there is not imported.